onchaind/onchain.c

```diff
@@ -509,6 +510,74 @@ static size_t num_not_irrevocably_resolved(struct tracked_output **outs)
 	return num;
 }
 
+static u32 prop_blockheight(const struct tracked_output *out)
+{
+	return out->tx_blockheight + out->proposal->depth_required;
+}
+
+static void billboard_update(struct tracked_output **outs)
+{
+	const struct tracked_output *best = NULL;
+
+	/* Highest priority is to report on proposals we have */
+	for (size_t i = 0; i < tal_count(outs); i++) {
+		if (!outs[i]->proposal)
+			continue;
+		if (!best || prop_blockheight(outs[i]) < prop_blockheight(best))
+			best = outs[i];
+	}
+
+	if (best) {
+		/* If we've broadcast and not seen yet, this happens */
+		if (best->proposal->depth_required <= best->depth) {
+			peer_billboard(false,
+				       "%u outputs unresolved: waiting confirmation that we spent %s (%s:%u) using %s",
+				       num_not_irrevocably_resolved(outs),
+				       output_type_name(best->output_type),
+				       type_to_string(trc, struct bitcoin_txid,
+						      &best->txid),
+				       best->outnum,
+				       tx_type_name(best->proposal->tx_type));
+		} else {
+			peer_billboard(false,
+				       "%u outputs unresolved: in %u blocks will spend %s (%s:%u) using %s",
+				       num_not_irrevocably_resolved(outs),
+				       best->proposal->depth_required - best->depth,
+				       output_type_name(best->output_type),
+				       type_to_string(trc, struct bitcoin_txid,
+						      &best->txid),
+				       best->outnum,
+				       tx_type_name(best->proposal->tx_type));
+		}
+		return;
+	}
+
+	/* Now, just report on the last thing we're waiting out. */
+	for (size_t i = 0; i < tal_count(outs); i++) {
+		/* FIXME: Can this happen?  No proposal, no resolution? */
+		if (!outs[i]->resolved)
+			continue;
+		if (!best || outs[i]->resolved->depth < best->resolved->depth)
+			best = outs[i];
+	}
+
+	if (best) {
+		peer_billboard(false,
```

Comment:
Because the perm line is still saying whether it's a mutual close, their unilateral, our unilateral, etc.  This one updates as the countdown goes down...